server/src/main/java/io/deephaven/server/arrow/FlightServiceGrpcImpl.java

```diff
@@ -109,6 +131,18 @@ public void onNext(final Flight.HandshakeRequest value) {
                 auth = login(BasicAuthMarshaller.AUTH_TYPE, protocolVersion, payload, handshakeResponseListener);
                 if (auth.isEmpty()) {
                     final WrappedAuthenticationRequest req = WrappedAuthenticationRequest.parseFrom(payload);
+                    // If the auth request is bearer, the v1 auth might be trying to renew an existing session
+                    if (req.getType().equals(Auth2Constants.BEARER_PREFIX.trim())) {
+                        try {
+                            UUID uuid = UuidCreator.fromString(req.getPayload().toString(StandardCharsets.US_ASCII));
+                            SessionState session = sessionService.getSessionForToken(uuid);
+                            respondWithAuthTokenBin(session);
+                            return;
+                        } catch (IllegalArgumentException | InvalidUuidException ignored) {
```

Comment:
Do we expect IllegalArgumentException to ever get hit?

Reply:
Looks like this one is a real one - if `session` were null, `respondWithAuthTokenBin` would NPE (which inherts from IAE).

Probably better to wrap in a null check, in conjunction with the change below.

Reply:
Okay, there's another way IAE can get hit here - if the req's payload isn't valid ascii, the String constructor will throw.

This is only an error in terms of "we cant read it as a uuid", this might be valid non-ascii bytes for some other auth handler.